program/src/state.rs

```diff
@@ -640,16 +628,15 @@ pub struct Validator {
     /// Sum of the balances of the stake accounts.
     pub stake_accounts_balance: Lamports,
 
-    /// Weight of the validator. Used when calculating the stake amount for
-    /// keeping a weighted balance, also defines the validator's share of fees.
-    pub weight: Weight,
+    /// Controls if a validator is allowed to have new stake deposits.
+    /// When removing a validator, this flag should be set to `true`.
+    pub inactive: bool,
```

Comment:
With booleans, it is often simpler to avoid a double negation, then you don’t have to work all of that out in your head. In this case, we can change the field to `active: bool`, and have it default to true.

Some evidence for this, is that almost all places in the code where this field is referenced, now write `!validator.inactive`, and then you have to mentally make the step “not inactive ⇒ active”. It would be a bit simpler to just write `validator.active` in those places.

Reply:
I originally did it like that, but then I'd lose the default `entry` constructor from `AccountMap`. But we could also implement the custom default for a `Validator`.

Reply:
:+1: Let’s implement `Default` manually for `Validator` then.